### ISSUE TITLE
Compiler crashes while building the connector code generated by openapi tool

### ISSUE THREAD

**Issue by kapilraajP** [ballerina-platform/ballerina-lang#31631]
**Description:**
While building an openapi generated ballerina connector. during ballerina build process compiler crashes with error log as follows
```
[2021-07-08 02:34:30,202] SEVERE {b7a.log.crash} - cannot find function: '$populate$type$Client' 
java.lang.IllegalStateException: cannot find function: '$populate$type$Client'
	at org.wso2.ballerinalang.compiler.bir.codegen.JvmPackageGen.findFunction(JvmPackageGen.java:369)
	at org.wso2.ballerinalang.compiler.bir.codegen.JvmPackageGen.findFunction(JvmPackageGen.java:353)
	at org.wso2.ballerinalang.compiler.bir.codegen.JvmPackageGen.getBytes(JvmPackageGen.java:651)
	at org.wso2.ballerinalang.compiler.bir.codegen.JvmPackageGen.lambda$generateModuleClasses$0(JvmPackageGen.java:453)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base/java.util.HashMap$EntrySpliterator.forEachRemaining(HashMap.java:1746)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.ForEachOps$ForEachTask.compute(ForEachOps.java:290)
	at java.base/java.util.concurrent.CountedCompleter.exec(CountedCompleter.java:746)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinTask.doInvoke(ForkJoinTask.java:408)
	at java.base/java.util.concurrent.ForkJoinTask.invoke(ForkJoinTask.java:736)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateParallel(ForEachOps.java:159)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateParallel(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:233)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
	at java.base/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:661)
	at org.wso2.ballerinalang.compiler.bir.codegen.JvmPackageGen.generateModuleClasses(JvmPackageGen.java:388)
	at org.wso2.ballerinalang.compiler.bir.codegen.JvmPackageGen.generate(JvmPackageGen.java:777)
	at org.wso2.ballerinalang.compiler.bir.codegen.CodeGenerator.generate(CodeGenerator.java:89)
	at org.wso2.ballerinalang.compiler.bir.codegen.CodeGenerator.generate(CodeGenerator.java:67)
	at io.ballerina.projects.JBallerinaBackend.performCodeGen(JBallerinaBackend.java:278)
	at io.ballerina.projects.ModuleContext.generateCodeInternal(ModuleContext.java:407)
	at io.ballerina.projects.ModuleCompilationState$4.generatePlatformSpecificCode(ModuleCompilationState.java:132)
	at io.ballerina.projects.ModuleContext.generatePlatformSpecificCode(ModuleContext.java:333)
	at io.ballerina.projects.JBallerinaBackend.performCodeGen(JBallerinaBackend.java:147)
	at io.ballerina.projects.JBallerinaBackend.<init>(JBallerinaBackend.java:133)
	at io.ballerina.projects.JBallerinaBackend.lambda$from$0(JBallerinaBackend.java:107)
	at java.base/java.util.HashMap.computeIfAbsent(HashMap.java:1133)
	at io.ballerina.projects.PackageCompilation.getCompilerBackend(PackageCompilation.java:152)
	at io.ballerina.projects.JBallerinaBackend.from(JBallerinaBackend.java:106)
	at io.ballerina.cli.task.CompileTask.execute(CompileTask.java:69)
	at io.ballerina.cli.TaskExecutor.executeTasks(TaskExecutor.java:40)
	at io.ballerina.cli.cmd.BuildCommand.execute(BuildCommand.java:272)
	at java.base/java.util.Optional.ifPresent(Optional.java:183)
	at io.ballerina.cli.launcher.Main.main(Main.java:58)
```

**Steps to reproduce:**

**Affected Versions:**

**OS, DB, other environment details and versions:**

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

**Comment by warunalakshitha** [comment]
This issue causes when we have a large amount of functions defined inside an object.

**Comment by LakshanSS** [comment]
Is the problem large number of operations or large number of lines?

**Comment by sanjiva** [comment]
@hasithaa please review this issue.

**Comment by warunalakshitha** [comment]
Update on the current status of the issue

Issue was because of connector code contained a client object with around 300 remote functions. During runtime initialization we create a method type for each remote functions and given "populateTypes" method cause method too large issue. We found few place we can optimize type generations logic.

We are working all those cases on splitting our type generation logic and also there are possibility optimizing the generated code. We have started work on that and hope to finish by this sprint (beta 3)
